PermissionsScraper/PermissionsScraper/Triggers/DescriptionsScraper.cs

```diff
@@ -0,0 +1,207 @@
+                        PopulateScopesDescriptions(permsAppConfig, authResult, apiVersion);
+                        log.LogInformation($"Finished fetching Service Principal permissions descriptions for {apiVersion}. Time: {DateTime.UtcNow}");
+                    }
+                }
+
+                if (!_scopesDescriptions.Any())
+                {
+                    log.LogInformation($"{nameof(_scopesDescriptions)} dictionary returned empty data. " +
+                        $"Exiting function DescriptionsScraper Time: {DateTime.UtcNow}");
+                    return;
+                }
+
+                var servicePrincipalScopes = JsonConvert.SerializeObject(_scopesDescriptions, Formatting.Indented);
+
+                // Fetch permissions descriptions from GitHub repo
+                var gitHubAppConfig = new GitHubRepoAppConfig
+                {
+                    GitHubAppId = permsAppConfig.GitHubAppId,
+                    GitHubOrganization = permsAppConfig.GitHubOrganization,
+                    GitHubAppName = permsAppConfig.GitHubAppName,
+                    GitHubRepoName = permsAppConfig.GitHubRepoName,
+                    ReferenceBranch = permsAppConfig.ReferenceBranch,
+                    FileContentPath = permsAppConfig.FileContentPath,
+                    WorkingBranch = permsAppConfig.WorkingBranch,
+                    Reviewers = permsAppConfig.Reviewers,
+                    PullRequestTitle = permsAppConfig.PullRequestTitle,
+                    PullRequestBody = permsAppConfig.PullRequestBody,
+                    PullRequestLabels = permsAppConfig.PullRequestLabels,
+                    PullRequestAssignees = permsAppConfig.PullRequestAssignees,
+                    CommitMessage = permsAppConfig.CommitMessage,
+                    TreeItemMode = Enums.TreeItemMode.Blob
+                };
+
+                log.LogInformation($"Fetching permissions descriptions from GitHub repository '{gitHubAppConfig.GitHubRepoName}'. Time: {DateTime.UtcNow}");
+
+                var repoScopes = BlobContentReader.ReadRepositoryBlobContentAsync(gitHubAppConfig, permsAppConfig.GitHubAppKey).GetAwaiter().GetResult();
+
+                log.LogInformation($"Finished fetching permissions descriptions from GitHub repository '{gitHubAppConfig.GitHubRepoName}'. Time: {DateTime.UtcNow}");
+
+                log.LogInformation($"Comparing scopes from the Service Principal and the GitHub repository '{gitHubAppConfig.GitHubRepoName}' " +
+                    $"for new updates... Time: {DateTime.UtcNow}");
+
+                // Compare GitHub permissions descriptions to Service Principal permissions descriptions
+                if (servicePrincipalScopes.Equals(repoScopes, StringComparison.OrdinalIgnoreCase))
+                {
+                    log.LogInformation($"No permissions descriptions update required. Exiting function 'DescriptionsScraper'. Time: {DateTime.UtcNow}");
```

Comment:
nit: you can have a return after this line to remove else block. That's what I meant to achieve by inverting if ;)

Reply:
I mean, we will need the contents of the else block, just the `else { }` around it can be removed for less indentation

Reply:
Yea, I gotch'a